lib/ethpy/ethpy/hyperdrive/api/api.py

```diff
@@ -482,12 +484,42 @@ async def async_redeem_withdraw_shares(
         """
         return await _async_redeem_withdraw_shares(self, agent, trade_amount, nonce)
 
+    def calc_open_long(self, base_amount: FixedPoint) -> FixedPoint:
+        """Calculate the long amount that will be opened for a given base amount after fees.
+
+        Arguments
+        ---------
+        base_amount : FixedPoint
+            The amount to spend, in base.
+
+        Returns
+        -------
+        long_amount : FixedPoint
+            The amount of bonds purchased.
+        """
+        return _calc_long_amount(self, base_amount)
+
+    def calc_open_short(self, short_amount: FixedPoint) -> FixedPoint:
+        """Calculate the amount of base the trader will need to deposit for a short of a given size.
+
+        Arguments
+        ---------
+        short_amount : FixedPoint
+            The amount to of bonds to short.
+
+        Returns
+        -------
+        short_amount : FixedPoint
+            The amount of base required to short the bonds (aka the "max loss").
+        """
+        return _calc_short_deposit(self, short_amount, self.spot_price, self.pool_info["sharePrice"])
```

Comment:
This seems the only place where this call is dependent on the internal state of the interface. I'd like the option to pass in my own pool information when calling rust's open short.

Reply:
yes, agreed. I did it this way to keep things consistent. I have plans for a follow-up API refactor that facilitates ensuring everything is synced across pool info, block, checkpoint. This refactor will also include the ability to pass your own pool info. https://github.com/delvtech/elf-simulations/issues/1022